Src/EB/AMReX_EBToPVD.cpp

```diff
@@ -0,0 +1,495 @@
+
+                  for(int idim = 0; idim < 3; ++idim) {
+                     apoints[ 0][idim] = vertex[0][idim] + ihat[idim]*dx[0]*alpha[ 0];
+                     apoints[ 1][idim] = vertex[1][idim] + jhat[idim]*dx[1]*alpha[ 1];
+                     apoints[ 2][idim] = vertex[2][idim] + ihat[idim]*dx[0]*alpha[ 2];
+                     apoints[ 3][idim] = vertex[0][idim] + jhat[idim]*dx[1]*alpha[ 3];
+                     apoints[ 4][idim] = vertex[0][idim] + khat[idim]*dx[2]*alpha[ 4];
+                     apoints[ 5][idim] = vertex[1][idim] + khat[idim]*dx[2]*alpha[ 5];
+                     apoints[ 6][idim] = vertex[3][idim] + khat[idim]*dx[2]*alpha[ 6];
+                     apoints[ 7][idim] = vertex[2][idim] + khat[idim]*dx[2]*alpha[ 7];
+                     apoints[ 8][idim] = vertex[4][idim] + ihat[idim]*dx[0]*alpha[ 8];
+                     apoints[ 9][idim] = vertex[5][idim] + jhat[idim]*dx[1]*alpha[ 9];
+                     apoints[10][idim] = vertex[6][idim] + ihat[idim]*dx[0]*alpha[10];
+                     apoints[11][idim] = vertex[4][idim] + jhat[idim]*dx[1]*alpha[11];
+                  }
+
+                  // store intersections with grid cell alpha in [0,1]
+                  for(int lc1 = 0; lc1 < 12; ++lc1) {
+                     if(alpha_intersect[lc1]) {
+                        m_points.push_back(apoints[lc1]);
+                        int lc2 = m_connectivity.back()[0]+1;
+                        m_connectivity.back()[0] = lc2;
+                        m_connectivity.back()[lc2] = m_points.size()-1;
+                     }
+                  }
+
+                  reorder_polygon(m_points, m_connectivity.back(), n0);
+               }
+            }
+         }
+      }
+   };
+}
+
+void EBToPVD::WriteEBVTP(const int myID) const
+{
+   std::stringstream ss;
+   ss << std::setw(8) << std::setfill('0') << myID;
+   std::string cID = "eb_" + ss.str() + ".vtp";
+
+   std::ofstream myfile(cID);
+   if(myfile.is_open()) {
+      myfile.precision(6);
+      myfile << "<?xml version=\"1.0\"?>" << std::endl;
+      myfile << "<VTKFile type=\"PolyData\" version=\"0.1\" byte_order=\"LittleEndian\">" << std::endl;
+      myfile << "<PolyData>" << std::endl;
+      myfile << "<Piece NumberOfPoints=\"" << m_points.size() << "\" NumberOfVerts=\"0\" "
+         << "NumberOfLines=\"0\" NumberOfString=\"0\" NumberOfPolys=\" "
+         << m_connectivity.size() << "\">" << std::endl;
+      print_points(myfile);
+      print_connectivity(myfile);
+      myfile << "<PointData></PointData>" << std::endl;
+      myfile << "<CellData></CellData>" << std::endl;
+      myfile << "</Piece>" << std::endl;
+      myfile << "</PolyData>" << std::endl;
+      myfile << "</VTKFile>" << std::endl;
+
+      myfile.close();
+   }
+}
+
+void EBToPVD::WritePVTP(const int nProcs) const
+{
+   std::ofstream myfile("eb.pvtp");
+
+   if(myfile.is_open()) {
+      myfile << "<?xml version=\"1.0\"?>" << std::endl;
```

Comment:
It will be a little faster if `std::endl` here and many other places are replaced with `"\n"`, because `std::endl` flushes the buffer.